### PR TITLE
修復 header issues

### LINKED ISSUE #40: Pending header tasks
There are several issues that still need to be fixed, but the functionality works fine on desktop. The developer will address these issues later:

- [ ] The mobile version will be implemented after the shadcn `Dialog` component is integrated into the search bar.
- [ ] Some spacing issues will be tweaked later.
- [ ] The missing hover effect on the menu items will be added later.

### PR DESCRIPTION
**Pull request by zim314**
- [x]  fix the padding of the hamburger component.
- [x]  fix the hover effect to change the color of the menu items. 
- [x]  fix header height.
- [x]  fix font weight is incorrect on date.
- [ ]  the mobile version will be implemented after the shadcn Dialog component is integrated into the search bar
  - for the mobile version, create a new feature branch to implement it.
- [x] convert px to rem.  
- [x] remove border.
- [x] change border and text color.
- [x] create hook to close the menu.

modify the menu design to be scrollable vertically.

closes #40

### CODE REVIEW

**Review by ellallu0903 — APPROVED** [review]
LGTM👍

**Comment by blp100** [comment]
header spacing is incorrect

Screenshot 2024-06-06 at 10 48 54 PM

File: src/components/Header/HamburgerBotton.tsx [hunk 1]
```diff
@@ -44,16 +46,16 @@ const HamburgerBotton = ({ isLoggedin, onSignOut }: Props) => {
         />
       </PopoverTrigger>
       <PopoverContent className="h-0 w-0 border-none bg-transparent p-0">
-        <div className="no-scrollbar absolute right-[-56px] hidden h-fit max-h-[450px] w-[272px] overflow-scroll rounded-[32px] border-4 border-black bg-surface pt-6 xl:block">
+        <div className="no-scrollbar absolute right-[-56px] box-content hidden h-fit max-h-[462px] w-[272px] overflow-scroll rounded-[32px] border-4 border-black bg-surface pt-6 xl:block">
```
Comment (ellallu0903) [inline src/components/Header/HamburgerBotton.tsx:49]:
```suggestion
        <div className="no-scrollbar absolute right-[-56px] box-border hidden h-fit max-h-[462px] w-[272px] overflow-scroll rounded-[32px] border-4 border-primary bg-surface pt-6 xl:block">
```

File: src/components/Header/HamburgerBotton.tsx [hunk 2]
```diff
@@ -44,16 +46,16 @@ const HamburgerBotton = ({ isLoggedin, onSignOut }: Props) => {
...
           {isLoggedin ? (
             <>
               {userList.map((user: List) => (
                 <Link
-                  className="mb-4 flex justify-between px-8 py-2"
+                  className="mb-4 flex justify-between px-8 py-2 hover:bg-primary/[0.03]"
```
Comment (ellallu0903) [inline src/components/Header/HamburgerBotton.tsx:54]:
```suggestion
                  className="mb-4 flex justify-between px-8 py-[0.625rem] hover:bg-primary/[0.03]"
```

**Review by ellallu0903 — COMMENTED** [review]
整體需要調整的地方：
1. px -> rem
2. color
3. padding & margin

File: src/components/Header/EmailButton.tsx [hunk 3]
```diff
@@ -52,10 +52,9 @@ const EmailButton = () => {
       <PopoverTrigger className="ml-[3px] mr-[0.8125rem] rounded-full border border-primary p-5 data-[state=open]:ml-0 data-[state=open]:mr-[0.625rem] data-[state=open]:border-4">
```
[the remark this answers is not in the document]
Comment (blp100) [reply src/components/Header/EmailButton.tsx:52]: 幫忙回應，這邊ml設定成px，是因為border 太粗會彈跳所設計的，所以才用px設定

[the remark this answers is not in the document]
Comment (ellallu0903) [reply src/components/Header/EmailButton.tsx:52]: 了解～那我這邊先 resolve

File: src/components/Header/index.tsx
[the remark this answers is not in the document]
Comment (zim314) [reply src/components/Header/index.tsx:1]: header 有 border 寫太習慣 不小心就加上去了 😂

**Comment by ellallu0903** [comment]
樣式調整 OK，選單點擊後可以關閉，但發現點選空白處或是選單之間的間距也會關閉🤣

File: src/components/Header/NotificationButton.tsx [hunk 4]
```diff
@@ -63,16 +64,17 @@ const NotificationButton = () => {
...
-          <Separator className="h-[1px] bg-primary" />
+          <Separator className="h-[0.0625rem] bg-primary" />
           <Link
             href="/"
-            className="flex h-[4.5rem] w-full items-center justify-center"
+            ref={triggerRef as React.RefObject<HTMLAnchorElement>}
+            className="flex h-[4.5rem] w-full items-center justify-center hover:bg-primary/[0.03]"
           >
             查看所有通知
           </Link>
```
Comment (fufuShih) [inline src/components/Header/NotificationButton.tsx:80]: 似乎太複雜，應該不用`useEffect`和`useRef`去監聽是否關閉，改成用`useRouter`和`useState`就能達成。大概如下，應該就能達到:
```suggestion
          <Button
            variant="ghost"
            onClick={() => {
              setIsOpen((prev) => !prev);
            }}
            asChild
            className="flex h-[4.5rem] w-full items-center justify-center hover:bg-primary/[0.03]"
          >
            <Link href="/">查看所有通知</Link>
          </Button>
```

Comment (fufuShih) [reply src/components/Header/NotificationButton.tsx:80]: 或是更簡單的:
```suggestion
          <Link
            href="/"
            onClick={() => {
              setIsOpen((prev) => !prev);
            }}
            className="flex h-[4.5rem] w-full items-center justify-center"
          >
            查看所有通知
          </Link>
```

**Review by fufuShih — APPROVED** [review]

**Pull request merged**